src/rust/src/lib.rs

```diff
@@ -487,36 +461,3 @@ pub(crate) fn invoke_host_function(
         )
     }
 }
-
-pub(crate) fn preflight_host_function(
```

Comment:
Would it be fine if we kept the preflight methods and maybe move them to tests? I use this in the tests for my  ConservationOfLumens [PR](https://github.com/stellar/stellar-core/pull/3703/commits/9ae8db301cca6d76ce63d23e8d36af58a32d9369#diff-40b7e6088e6fc4c86852713558159ac030d1021a513832476da99221c4b716b6) to call into the token contract. In general, preflight would be helpful in test cases so we can call into more complex contracts in enforcing mode without building the footprint.

Reply:
I'd be willing to preserve it, I don't feel strongly besides ... well ... it's code and I like deleting code, and it's also code that runs synchronously so having it potentially exposed on a CommandHandler path is a bit unfortunate also. But I could go either way. I know @dmkozh and @MonsieurNicolas have opinions, perhaps they could weigh in?

Reply:
I don't really want preflight as a concept to stay in core because there are no external users of it. The current implementation comes with a burden of maintaining two very similar code paths (which are already somewhat divergent).
If we want test utils, let's introduce test utils. E.g. if you don't really care about the footprint, why not switch host to the recording mode for tests and just run the transaction as is (this could be achieved by just passing an additional parameter to a host function call as opposed to the completely separate code path).

Reply:
Yeah I meant moving them into test utils, not gating the current code behind `BUILD_TESTS`. I also wanted to use enforcing mode because I think that path in the rust bridge should be tested as well. Regardless of what we do though, I think we can merge this and I'll make the changes I need in my ConservationOfLumens PR.

Reply:
I'm not sure why we should cherry-pick e.g. footprints as a part of unrelated logic to test and not e.g. ledger close logic. Both are more or less equally unrelated to the logic you're trying to cover.

With introduction of resource limits the core tests that just want to call a contract for the sake of covering some business logic to be performed would become even more daunting to setup (even with 'preflight'-style logic), as you need to fill not one, but several fields in the transaction, none of which really have anything to do with the logic you're trying to test. That's why I feel like a recording mode is suitable for most of the tests, besides the 'e2e' style tests where you're really interested in footprints etc. being taken into account.

Reply:
Another thing I don't like much about building the test inputs implicitly (like in the case of using the 'preflight' footprint) is that they don't uncover much. E.g. imagine a bug that causes us to access unexpected ledger entry. Preflight would happily return it to you and you'll never know that something is wrong. If you really wanted to make sure that the expected entries are being accessed you have to pass them to the test explicitly. If you're not doing that, then your coverage is roughly as good as the recording mode (at which point why not just use the recording mode and save some test run time and simplify the test code).

Reply:
Yeah I agree that using recording mode to test enforcing mode isn't perfect. So you're proposing to call into the `rust_bridge` directly in the test instead of using `InvokeHostFunctionOp`? This actually wouldn't work in my case because I need to use the operation to trigger the invariant. The other option would be to compile in an option somehow to enabled recording mode in tests. For now, It might be easier if I just manually pass the footprint, but we'll have to address this problem at some point.

Reply:
I updated my tests to not using prefetching for now - https://github.com/stellar/stellar-core/pull/3703/commits/00d5263df517f9450b7b1330c386af1b283b376f

Reply:
Thanks, that's good for now and we can figure out more holistic testing approach when/if needed.